components/BarcodeScanner.vue

```diff
@@ -202,6 +215,10 @@ export default {
         await this.scanner.open();
       } catch(e) {
         console.error(e);
+        if(e.message === 'Requested device not found') {
+          this.cameraError = true;
```

Comment:
We want to have this special error handling instead of using the `this.$emit` infrastructure?

Reply:
In this case yes because if the camera is not available, the user still needs to have the option to upload a photo through the scanner.

Reply:
OK